crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -569,6 +569,15 @@ fn handle_own_line_comment_between_statements<'a>(
     // y = 2
     // ```
     if max_empty_lines(&source[TextRange::new(comment.end(), following.start())]) == 0 {
+        for token in SimpleTokenizer::new(source, TextRange::new(comment.end(), following.start()))
```

Comment:
Can you say more why we're handling this case here? I'd expect this to be handled by `handle_own_line_comment_between_branches`